python/knowledges/batch_upload/delete_duplicate_files.py

```
@@ -0,0 +1,211 @@
+                print(f"❌ Failed: {filename} - {e}")
+    
+    # Save deletion log
+    deletion_log = {
+        'timestamp': datetime.now().isoformat(),
+        'knowledge_base_id': KNOWLEDGE_BASE_ID,
+        'integrity_report_used': INTEGRITY_REPORT_PATH,
+        'total_files': len(duplicate_files),
+        'successful_deletions': len(deleted_files),
+        'failed_deletions': len(failed_deletions),
+        'deleted_files': deleted_files,
+        'failed_deletions': failed_deletions
```

Ruff (0.12.2)

Dictionary key literal `'failed_deletions'` repeated

(F601)